static/survey-ad.js

```diff
@@ -0,0 +1,50 @@
+var Control = require("can-control");
+
+module.exports = Control.extend({
+  /* This should only happen once */
+  init: function() {
+    var surveyAdElement = this.element;
+
+    // Always show the close button since JS is active
+    var closeButton = surveyAdElement.querySelector('.close');
+    if (closeButton.classList) {// Only enable the close button in IE10+
+      closeButton.style.display = 'inline-block';
+    }
+
+    // Look up the user’s preference in localStorage
+    try {
+      var storageKey = 'survey-ad-closed';
+      var didClose = window.localStorage.getItem(storageKey);
+      if (didClose) {
+        // Immediately hide this element
+        surveyAdElement.style.display = 'none';
```

Comment:
Have you considered hiding by default and showing if needed?

Reply:
Yeah, I would rather have the link be shown when JS is disabled or `localStorage` is unavailable than hide it in those cases.